typegate/native/src/runtimes/prisma/engine.rs

```diff
@@ -16,18 +16,17 @@ pub async fn register_engine(datamodel: String, tg_name: String) -> Result<Strin
         ignore_env_var_errors: false,
     };
     let engine = super::engine_import::QueryEngine::new(conf)
-        .with_context(|| format!("Error while registering engine for typegraph {tg_name}"))?;
+        .with_context(|| format!("Error while registering engine {engine_name}"))?;
     engine.connect().await?;
-    let engine_id = format!("{tg_name}_{}", super::ENGINES.len() + 1);
```

Comment:
bug here, dash map.len is not atomic registering same engine multiple times on fast computer

Reply:
Nice catch, that will *definitely* happen.